srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -195,19 +209,28 @@ private void retrieveLabs(final int dfn, final Patient patient) throws ParseExce
         final VistaLabs[] enumValues = VistaLabs.values();
         for(final VistaLabs labRetrievalEnum: enumValues)
         {
-            final String rpcResultString = fProcedureCaller.doRetrieveLabs(
-                    fDuz,
-                    String.valueOf(dfn),
-                    labRetrievalEnum.getPossibleLabNames());
-            // If the resultString is a success, add it to the patient's lab data.
-            // Else, we don't need to do anything.
-            if(!rpcResultString.isEmpty())
+            try
             {
-                final String[] rpcSplit = rpcResultString.split("\\^");
-                final double labValue = Double.parseDouble(rpcSplit[1]);
-                final SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy@HH:mm:ss");
-                patient.getLabs().put(labRetrievalEnum.name(),
-                        new RetrievedValue(labValue, format.parse(rpcSplit[2]), rpcSplit[3]));
+                final String rpcResultString = fProcedureCaller.doRetrieveLabs(
+                        fDuz,
+                        String.valueOf(dfn),
+                        labRetrievalEnum.getPossibleLabNames());
+                // If the resultString is a success, add it to the patient's lab data.
+                // Else, we don't need to do anything.
+                if(!rpcResultString.isEmpty())
+                {
+                    List<String> rpcSplit = Splitter.on('^').splitToList(rpcResultString);
+                    final double labValue = Double.parseDouble(rpcSplit.get(1));
+                    final SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy@HH:mm:ss");
+                    patient.getLabs().put(labRetrievalEnum.name(),
+                            new RetrievedValue(labValue, format.parse(rpcSplit.get(2)), rpcSplit.get(3)));
+                }
+            }
+            catch(final Exception e)
+            {
+                // If an exception occurs for any reason, move to the next lab so that as much patient
+                // data as possible can still be retrieved.
+                fLogger.debug("Unable to retrieve lab {}. {}", labRetrievalEnum.name(), e.toString());
```

Comment:
I can see why you made this DEBUG-level, but I think WARN is more appropriate per the standards in the dev guide. This is definitely an exceptional condition and really shouldn't happen in production at all.